src/app/InteractionModelEngine.cpp

```diff
@@ -427,54 +513,6 @@ Protocols::InteractionModel::Status InteractionModelEngine::OnReadInitialRequest
         }
     }
 
-    if (aInteractionType == ReadHandler::InteractionType::Read)
-    {
-        System::PacketBufferTLVReader reader;
-        reader.Init(aPayload.Retain());
-
-        ReadRequestMessage::Parser readRequestParser;
-        VerifyOrReturnError(readRequestParser.Init(reader) == CHIP_NO_ERROR, Status::InvalidAction);
-
-        {
-            size_t requestedAttributePathCount = 0;
-            size_t requestedEventPathCount     = 0;
-            AttributePathIBs::Parser attributePathListParser;
-            CHIP_ERROR err = readRequestParser.GetAttributeRequests(&attributePathListParser);
-            if (err == CHIP_NO_ERROR)
-            {
-                TLV::TLVReader pathReader;
-                attributePathListParser.GetReader(&pathReader);
-                ReturnErrorCodeIf(TLV::Utilities::Count(pathReader, requestedAttributePathCount, false) != CHIP_NO_ERROR,
-                                  Status::InvalidAction);
-            }
-            else if (err != CHIP_ERROR_END_OF_TLV)
-            {
-                return Status::InvalidAction;
-            }
-            EventPathIBs::Parser eventpathListParser;
-            err = readRequestParser.GetEventRequests(&eventpathListParser);
-            if (err == CHIP_NO_ERROR)
-            {
-                TLV::TLVReader pathReader;
-                eventpathListParser.GetReader(&pathReader);
-                ReturnErrorCodeIf(TLV::Utilities::Count(pathReader, requestedEventPathCount, false) != CHIP_NO_ERROR,
-                                  Status::InvalidAction);
-            }
-            else if (err != CHIP_ERROR_END_OF_TLV)
-            {
-                return Status::InvalidAction;
-            }
-
-            // The following cast is safe, since we can only hold a few tens of paths in one request.
-            Status checkResult = EnsureResourceForRead(apExchangeContext->GetSessionHandle()->GetFabricIndex(),
```

Comment:
Wait, what?  Why is `EnsureResourceForRead` being removed?  That's not right!

Reply:
just bring it up, thanks for pointing out this.